version: "3"
services:
  web:
    build: .
    restart: on-failure
    depends_on:
      - db
    links: 
      - db
    ports:
      - "3000:3000"
    volumes:
      - yprov_data:/app/conf

  db:
    image: "neo4j:enterprise"
    ports:
      - "7474:7474"
      - "7687:7687"
    environment:
      NEO4J_AUTH: neo4j/password
      NEO4J_ACCEPT_LICENSE_AGREEMENT: eval 
      NEO4J_PLUGINS: '["graph-data-science"]'
    volumes:
      - neo4j_data:/var/lib/neo4j/
      - neo4j_logs:/var/log/neo4j/
    healthcheck:
        test: ["CMD", "curl", "-f", "http://db:7474"]
        interval: 30s
        timeout: 10s
        retries: 5

volumes:
  neo4j_data:
    external: True
  neo4j_logs:
    external: True
  yprov_data:
    external: True